examples/chat/src/dialer.js

```diff
@@ -13,27 +13,28 @@ async function run() {
   ])
 
   // Create a new libp2p node on localhost with a randomly chosen port
-  const peerDialer = new PeerInfo(idDialer)
-  peerDialer.multiaddrs.add('/ip4/0.0.0.0/tcp/0')
   const nodeDialer = new Node({
-    peerInfo: peerDialer
+    peerId: idDialer,
+    addresses: {
+      listen: ['/ip4/0.0.0.0/tcp/0']
+    }
   })
 
-  // Create a PeerInfo with the listening peer's address
-  const peerListener = new PeerInfo(idListener)
-  peerListener.multiaddrs.add('/ip4/127.0.0.1/tcp/10333')
+  // Add the listener peer multiaddr to the PeerStore
+  // This would be automatic if it was discovered through a discovery service
+  nodeDialer.peerStore.addressBook.set(idListener, [multiaddr('/ip4/127.0.0.1/tcp/10333')])
 
   // Start the libp2p host
   await nodeDialer.start()
 
   // Output this node's address
   console.log('Dialer ready, listening on:')
-  peerListener.multiaddrs.forEach((ma) => {
+  nodeDialer.transportManager.getAddrs().forEach((ma) => {
```

Comment:
Use the multiaddrs getter
```suggestion
  nodeDialer.multiaddrs.forEach((ma) => {
```